yb-voyager/src/srcdb/data/gather-assessment-metadata/oracle/object-type-mapping.sqlplus

```diff
@@ -16,15 +16,21 @@ SPOOL object-type-mapping.csv
 SELECT schema_name, object_name, object_type
 FROM (
     SELECT
-        owner AS schema_name,
-        -- TODO: handle the partitioned table support later
-        object_name || NVL2(subobject_name, '_' || subobject_name, '') AS object_name,
-        object_type
+        o.owner AS schema_name,
+        CASE
+            WHEN o.object_type = 'TABLE' THEN o.object_name
+            WHEN o.object_type = 'TABLE PARTITION' THEN o.object_name || '_' || o.subobject_name
+            WHEN o.object_type = 'TABLE SUBPARTITION' THEN o.object_name || '_' || sp.partition_name || '_' || o.subobject_name
+            ELSE o.object_name || NVL2(o.subobject_name, '_' || o.subobject_name, '')
```

Comment:
THe NVL2 seems confusing, how do both the options have `o.subobject_name` ?

Reply:
The NVL2 function in Oracle is a conditional function that evaluates the first argument and returns one value if it is not null, and another value if it is null. The syntax for NVL2 is as follows:

NVL2(expr1, expr2, expr3)
- expr1: The expression to be checked for NULL.
- expr2: The value returned if expr1 is not NULL.
- expr3: The value returned if expr1 is NULL.